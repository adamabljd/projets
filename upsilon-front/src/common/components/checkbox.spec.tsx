import React from 'react';
import { fireEvent, render, RenderResult, screen } from '@testing-library/react';
import Checkbox, { CheckboxProps } from './checkbox.component';

const handleOnChangeMock = jest.fn();

const mockCheckboxProps: CheckboxProps = {
  onChange: handleOnChangeMock
};
function renderCheckbox(): RenderResult {
  return render(<Checkbox {...mockCheckboxProps} />);
}

describe('Checkbox', () => {
  describe('static rendering', () => {
    it('should render itself', () => {
      const { baseElement } = renderCheckbox();
      expect(baseElement).toMatchSnapshot();
    });
  });
  describe('checked', () => {
    it('should be checked', () => {
      render(<Checkbox {...mockCheckboxProps} />);
      const checkbox = screen.getByRole('checkbox');
      fireEvent.click(checkbox);
      expect(handleOnChangeMock).toHaveBeenCalled();
    });
    it('should return the value of checked', () => {
      render(<Checkbox {...mockCheckboxProps} />);
      const checkbox = screen.getByRole('checkbox') as HTMLInputElement;
      fireEvent.click(checkbox);
      expect(checkbox.checked).toBe(true);
    });
  });
});
