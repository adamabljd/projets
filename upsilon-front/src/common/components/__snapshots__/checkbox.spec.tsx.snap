// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox static rendering should render itself 1`] = `
<body>
  <div>
    <input
      class="checkbox-component"
      type="checkbox"
    />
  </div>
</body>
`;
